### ISSUE TITLE
jiralib2-session-login fails with [error] request--callback: JSON readtable error: 40

### ISSUE THREAD

**Issue by yamatakau08** [nyyManni/jiralib2#4]
My environment is
Msys2 mingw64 and emacs 
```
(emacs-version)
"GNU Emacs 26.3 (build 1, x86_64-w64-mingw32)
 of 2019-11-02"
```

When I execute jiralib2-session-login to interact with my corp JIRA, jiralib2 return "Request failed: Could not reach the server".

I dig where wrong is.
I found the following error message in request block.

**{"errorMessages":["Unexpected character ('u' (code 117)): expected a valid value ...**
**[error] request--callback: JSON readtable error: 40**

Is there any way how to avoid this error?

Supplement information.

in scratch buffer
```
request-curl
"curl"
(executable-find request-curl)
"c:/msys64/usr/bin/curl.exe"
(setq request-log-level 'debug)
debug
(setq request-message-level 'debug)
debug
```

The following is request full debug message.

```
[debug] request--curl: curl --silent --location --cookie c:/yama/.emacs.d/request/curl-cookie-jar --cookie-jar c:/yama/.emacs.d/request/curl-cookie-jar --include --write-out \n(:num-redirects %{num_redirects} :url-effective %{url_effective}) --compressed --data-binary @c:/Users/000091~1/AppData/Local/Temp/emacs-requestnRROyU --request POST --header Content-Type: application/json https://www.tool.mycorp.biz/jira3/rest/auth/1/session
[debug] request--curl-callback: event finished

[debug] request--callback: UNPARSED

(:num-redirects 0 :url-effective password:hogehoge)HTTP/1.1 200 Connection established



HTTP/2 400 

date: Tue, 14 Apr 2020 01:06:32 GMT

content-type: application/json;charset=UTF-8

content-length: 207

set-cookie: AWSALB=v7sHULgUTwH78j3aLHTXOmWAUc/YTk3BvABGp0Vo3TnHUMvnH9TH8Q2BJg4lTTnACUlQLw4k67eXcjmtsNDVdW+GIEqLqlqdaxwCNcGIx1Z9pMDUcEejkHz0aEqP; Expires=Tue, 21 Apr 2020 01:06:32 GMT; Path=/

set-cookie: AWSALBCORS=v7sHULgUTwH78j3aLHTXOmWAUc/YTk3BvABGp0Vo3TnHUMvnH9TH8Q2BJg4lTTnACUlQLw4k67eXcjmtsNDVdW+GIEqLqlqdaxwCNcGIx1Z9pMDUcEejkHz0aEqP; Expires=Tue, 21 Apr 2020 01:06:32 GMT; Path=/; SameSite=None; Secure

server: Apache

x-arequestid: 606x884703x2

x-xss-protection: 1; mode=block

x-content-type-options: nosniff

x-frame-options: SAMEORIGIN

content-security-policy: frame-ancestors 'self'

x-asen: SEN-2142834

x-asessionid: 1imrpu

x-ausername: anonymous

cache-control: no-cache, no-store, no-transform

content-encoding: gzip

vary: User-Agent

set-cookie: atlassian.xsrf.token=BYQ7-3M04-IV5R-DIUI_77ad03398c8a3fb26c858937396345390bda1a86_lout; Path=/jira3; Secure

set-cookie: JSESSIONID=E33C01D1FF3A9DF410F8189B59AC6844; Path=/jira3; Secure; HttpOnly



{"errorMessages":["Unexpected character ('u' (code 117)): expected a valid value (number, String, array, object, 'true', 'false' or 'null')\n at [Source: org.apache.catalina.connector.CoyoteInputStream@462591fe; line: 1, column: 2]"]}
[error] request--callback: JSON readtable error: 40
[debug] request--callback: executing error
[error] request-default-error-callback: https://www.tool.mycorp.biz/jira3/rest/auth/1/session parse-error
user-error: Request failed: Could not reach the server
Mark set [2 times]
Quit [2 times]
Mark set
```

**Comment by yamatakau08** [comment]
I switched request-backend to url-retrieve
`(setq request-backend 'url-retrieve)`
It works fine.

I still keep waiting for your reply on using curl of request-backend.

**Comment by yamatakau08** [comment]
I figured it out!

On msys2 mingw64 emacs environment,
When request-backend is curl, request's GET method has the following error and doesn't work.
```
{"errorMessages":["Unexpected character ('u' (code 117)): expected a valid value (number, String, array, object, 'true', 'false' or 'null')\n at [Source: org.apache.catalina.connector.CoyoteInputStream@461e5b9e; line: 1, column: 2]"]}
```
So, I tried to use url-retrieve as request-backend
```
(setq request-backend 'url-retrieve)
```
It works to a certain degree.

But GET/PUT has multibyte characters, have the following error
GET response data can't be displayed as readable character.
PUT have the following error
```
Debugger entered--Lisp error: (error \"error: (error Multibyte text in HTTP request: PUT https://www.tool.corp.biz/confluence/rest/api/content/...
```
I suspect it cause in curl command itsefl on msys2 with referring https://github.com/ahungry/org-jira/issues/49#issuecomment-303870919

Then, I downloaded curl command windows native from https://curl.haxx.se/dlwiz/
and specified it as request-curl command and revert request-backend to curl.

It WORKS FINE and SOLVED the avobe all cumbersome problem.
